File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/athena/src/main/java/aws/example/athena/CreateNamedQueryExample.java
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package aws.example.athena;

import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.athena.AthenaClient;
import software.amazon.awssdk.services.athena.model.AthenaException;
import software.amazon.awssdk.services.athena.model.CreateNamedQueryRequest;

/**
 * To run this Java V2 code example, ensure that you have setup your development environment, including your credentials.
 *
 * For information, see this documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */

public class CreateNamedQueryExample {
    public static void main(String[] args) throws Exception {

        final String USAGE = "\n" +
                "Usage:\n" +
                "    <name>\n\n" +
                "Where:\n" +
                "    name - the name of the Amazon Athena query. \n\n" ;

        if (args.length != 1) {
            System.out.println(USAGE);
            System.exit(1);
        }

        String name = args[0];
        AthenaClient athenaClient = AthenaClient.builder()
                .region(Region.US_WEST_2)
                .build();

        createNamedQuery(athenaClient, name);
        athenaClient.close();
    }

    public static void createNamedQuery(AthenaClient athenaClient, String name) {

        try {
            // Create the named query request.
            CreateNamedQueryRequest createNamedQueryRequest = CreateNamedQueryRequest.builder()
                    .database(ExampleConstants.ATHENA_DEFAULT_DATABASE)
                    .queryString(ExampleConstants.ATHENA_SAMPLE_QUERY)
                    .description("Sample Description")
                    .name(name)
                    .build();

            athenaClient.createNamedQuery(createNamedQueryRequest);
            System.out.println("Done");
        } catch (AthenaException e) {
            e.printStackTrace();
            System.exit(1);
        }
    }
}

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/athena/src/main/kotlin/com/kotlin/athena/CreateNamedQueryExample.kt

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.athena

import aws.sdk.kotlin.services.athena.AthenaClient
import aws.sdk.kotlin.services.athena.model.CreateNamedQueryRequest
import kotlin.system.exitProcess

suspend fun main(args:Array<String>) {

    val usage = """
    Usage:
        <queryString> <namedQuery> <database>

    Where:
        queryString - the query string to use (for example, "SELECT * FROM mydatabase"; ).
        namedQuery - the name of the query to create. 
        database - the name of the database to use (for example, mydatabase ).
        
    """

    if (args.size != 3) {
        println(usage)
        exitProcess(0)
    }

    val queryString = args[0]
    val namedQuery = args[1]
    val database = args[2]
    val id = createNamedQuery(queryString, namedQuery, database)
    println("The query ID is $id")
}

    suspend  fun createNamedQuery(queryStringVal:String, namedQuery:String, databaseVal:String ):String? {

            AthenaClient { region = "us-west-2" }.use { athenaClient->
            val resp =  athenaClient.createNamedQuery(CreateNamedQueryRequest {
                database = databaseVal
                queryString = queryStringVal
                description = "Created via the AWS SDK for Kotlin"
                this.name = namedQuery
            })
            return resp.namedQueryId
        }
    }
